src/Cake.Core/Tooling/ToolSettings.cs

```diff
@@ -12,5 +13,11 @@ public class ToolSettings
         /// </summary>
         /// <value>The tool path.</value>
         public FilePath ToolPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the arguments.
+        /// </summary>
+        /// <value>The arguments.</value>
+        public Action<ProcessArgumentBuilder> ArgumentCustomization { get; set; }
```

Comment:
Wonder if it would make sense to have it be a Func instead if Action instead 
```charp
public Func<ProcessArgumentBuilder, ProcessArgumentBuilder> ArgumentCustomization { get; set; }
```
so we could modify and return same instance, with the extension methods:
```csharp
Settings.ArgumentCustomization = arg=>arg.Append("-true");
```
,but also just as easily return a new instance/exsisting instance. 
```csharp
Settings.ArgumentCustomization = arg=>"my new arguments";
...
Settings.ArgumentCustomization = arg=>reUseExistingArguments;
...
Settings.ArgumentCustomization = arg=>GetCustomNewArgs(param1, param2);
```